src/MessagePack.UnityClient/Assets/Scripts/MessagePack/MessagePackCompression.cs

```diff
@@ -25,5 +25,33 @@ public enum MessagePackCompression
         /// See also ThisLibraryExtensionTypeCodes.LZ4
         /// </devremarks>
         Lz4Block,
+
+        /// <summary>
+        /// Compresses an entire msgpack sequence as a multi contiguous block.
+        /// This is the fastest and results in the best LZ4 compression,
+        /// at the cost of copying the entire sequence when necessary to get contiguous memory.
+        /// </summary>
+        /// <remarks>
+        /// Uses msgpack type code ext98 in array.
+        /// </remarks>
+        /// <devremarks>
+        /// See also ThisLibraryExtensionTypeCodes.LZ4Contiguous
+        /// </devremarks>
+        Lz4ContiguousBlock,
+    }
+
+#pragma warning disable SA1649 // File name should match first type name
```

Comment:
Why not just give the extensions class its own file?

Reply:
I think that separating files will be difficult to see.
Simple Enums and simple Extensions are easier to see closer.
This is my preference.